ambry-router/src/main/java/com/github/ambry/router/DeleteOperation.java

```diff
@@ -407,7 +414,38 @@ private boolean shouldRunOfflineRepair() {
         && routerConfig.routerDeleteOfflineRepairEnabled
         && operationTracker.getSuccessCount() > 0
         && getPrecedenceLevel(errorCode) >= getPrecedenceLevel(RouterErrorCode.AmbryUnavailable)
-        && (serviceId == null || !serviceId.startsWith(BackgroundDeleteRequest.SERVICE_ID_PREFIX)));
+        && !isServiceBackgroundDeleter());
+    // @formatter:on
+  }
+
+  private boolean shouldIssueForceDelete() {
+    // The conditions to issue force delete to replicas
+    // 1. the feature is enabled
+    // 2. there is no successful replica. If we have success replicas, we should try on demand replication and retry
+    // 3. error code precedence is over AmbryUnavailable. We don't do replication if the error is BlobExpired or TooManyRequests.
```

Comment:
nit: Seems like comment is referencing `replication`.

Reply:
updated